spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/startup/StartupTimeMetrics.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2012-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.metrics.startup;
+
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.TimeGauge;
+
+import org.springframework.boot.context.event.ApplicationStartedEvent;
+import org.springframework.context.ApplicationListener;
+
+/**
+ * Binds application startup metrics in response to an {@link ApplicationStartedEvent}.
+ *
+ * @author Chris Bono
+ * @since 2.6.0
+ */
+public class StartupTimeMetrics implements ApplicationListener<ApplicationStartedEvent> {
+
+	private final MeterRegistry meterRegistry;
+
+	private final Iterable<Tag> tags;
+
+	public StartupTimeMetrics(MeterRegistry meterRegistry) {
+		this(meterRegistry, Collections.emptyList());
+	}
+
+	public StartupTimeMetrics(MeterRegistry meterRegistry, Iterable<Tag> tags) {
+		this.meterRegistry = meterRegistry;
+		this.tags = (tags != null) ? tags : Collections.emptyList();
+	}
+
+	@Override
+	public void onApplicationEvent(ApplicationStartedEvent event) {
+		if (event.getStartupTime() == null) {
+			return;
+		}
+		TimeGauge
+				.builder("spring.boot.application.started", () -> event.getStartupTime().toMillis(),
```

Comment:
@mbazos I know you had mentioned the following possible [metric candidates](https://github.com/spring-projects/spring-boot/issues/26729#issuecomment-878315046) for a StartupTimeMetrics.

- jvm start time
- application start time
- spring context start time


### spring context start time

Just to clarify that the `StartupInfoLogger` is actually logging ["spring context start time"](https://github.com/spring-projects/spring-boot/blob/6c48e989b6140f4cac77cfc0d1f8ab25a7f3ad65/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/SpringApplication.java#L306) which corresponds to the `ApplicationStartedEvent` (the app is live but not ready for requests yet - it still has to invoke any runners). This pull request logs this startup time.

### application start time 
The ["application start time"](https://github.com/spring-projects/spring-boot/blob/6c48e989b6140f4cac77cfc0d1f8ab25a7f3ad65/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/SpringApplication.java#L317) I am guessing would correspond to the `ApplicationReadyEvent` (the app is live and the app is ready for requests - all runners have been invoked). This pull request does not log this startup time. It could easily be added though and it would allow users to get an idea how much their runners are adding to startup time.

### jvm start time
The `StartupInfoLogger` is logging ["jvm uptime"](https://github.com/spring-projects/spring-boot/blob/6c48e989b6140f4cac77cfc0d1f8ab25a7f3ad65/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/StartupInfoLogger.java#L95) which corresponds to how many ms the JVM has been up. This could also give an idea of the contribution the JVM has on the startup time. This pull request does not log this startup time either but it could easily be added. 

@wilkinsona @mbazos 
Wdyt about me adding in the 2 above metrics?